src/net/netsyscall.c

```diff
@@ -1888,6 +1888,43 @@ sysreturn setsockopt(int sockfd,
             goto unimplemented;
         }
         break;
+    case SOL_SOCKET:
+        switch (optname) {
+        case SO_REUSEADDR:
+            if (optlen != sizeof(int))
+                return -EINVAL;
+            if (s->sock.type == SOCK_STREAM) {
+                if (*((int *)optval))
+                    s->info.tcp.lw->so_options |= SOF_REUSEADDR;
+                else
+                    s->info.tcp.lw->so_options &= ~SOF_REUSEADDR;
```

Comment:
Alternatively to setting and clearing directly the option bits, ip_set_option() and ip_reset_option() could be called (same for UDP below).

Reply:
I switched over to use the lwip convenience functions!